samples/Owin.Nowin.HelloWorld/NowinServer.cs

```diff
@@ -0,0 +1,49 @@
+﻿using System.Net;
+using System.Threading.Tasks;
+using System.Collections.Generic;
+using Microsoft.AspNet.Builder;
+using Microsoft.AspNet.Hosting.Server;
+using Microsoft.AspNet.Http.Features;
+using Microsoft.AspNet.Http.Internal;
+using Microsoft.AspNet.Owin;
+using Nowin;
+
+namespace NowinWebSockets
+{
+    public class NowinServer : IServer
+    {
+        private RequestDelegate _callback;
+        private INowinServer _nowinServer;
+
+        public IFeatureCollection Features { get; }
+
+        private Task HandleRequest(IDictionary<string, object> env)
+        {
+            return _callback(new DefaultHttpContext(new OwinFeatureCollection(env)));
+        }
+
+        public NowinServer(IFeatureCollection features)
+        {
+            Features = features;
+        }
+
+        public void Start(RequestDelegate requestDelegate)
+        {
+            var builder = ServerBuilder.New()
```

Comment:
I would actually split this up. Do this configuration in CreateServer. The only job of IServer would be to call SetOwinApp and Start on the builder